src/statespace/SO2.cpp

```diff
@@ -73,53 +74,52 @@ std::size_t SO2::getStateSizeInBytes() const
 }
 
 //==============================================================================
-StateSpace::State* SO2::allocateStateInBuffer(void* _buffer) const
+StateSpace::State* SO2::allocateStateInBuffer(void* buffer) const
 {
-  return new (_buffer) State;
+  return new (buffer) State;
 }
 
 //==============================================================================
-void SO2::freeStateInBuffer(StateSpace::State* _state) const
+void SO2::freeStateInBuffer(StateSpace::State* state) const
 {
-  static_cast<State*>(_state)->~State();
+  static_cast<State*>(state)->~State();
 }
 
 //==============================================================================
 void SO2::compose(
-    const StateSpace::State* _state1,
-    const StateSpace::State* _state2,
-    StateSpace::State* _out) const
+    const StateSpace::State* state1,
+    const StateSpace::State* state2,
+    StateSpace::State* out) const
 {
-  // TODO: Disable this in release mode.
-  if (_state1 == _out || _state2 == _out)
-    throw std::invalid_argument("Output aliases input.");
+#ifndef NDEBUG // Debug mode
+  assert(state1 != out && state2 != out);
```

Comment:
I don't know why we wanted to disable this in release mode, but it seems we need this precondition check (using exception) in both modes according to the [docstring](https://github.com/personalrobotics/aikido/blob/master/include/aikido/statespace/StateSpace.hpp#L92-L93).

Reply:
Good catch :+1: